lib/cdo/i18n_string_url_tracker.rb

```diff
@@ -11,13 +12,44 @@ class I18nStringUrlTracker
   # @param url [String] The url which required the translation of the given string_key.
   # @param source [String] Context about where the string lives e.g. 'ruby', 'maze', 'turtle', etc
   def log(string_key, url, source)
-    return unless string_key && url && source
     return unless DCDO.get(I18N_STRING_TRACKING_DCDO_KEY, false)
 
+    url = normalize_url(url)
+    return unless string_key && url && source
+
     # record the string : url association.
     FirehoseClient.instance.put_record(
       :i18n,
       {url: url, string_key: string_key, source: source}
     )
   end
+
+  private
+
+  def normalize_url(url)
+    return unless url
+
+    parsed_url = URI(url)
+
+    # strip query string
+    parsed_url.query = nil
+
+    # strip anchor tags
+    parsed_url.fragment = nil
+
+    # Aggregate /projects/<project_type>/<project_id> URLs because each project URL will have a unique identifier
+    # converts 'https://studio.code.org/projects/flappy/zjiufOp0h-9GS-DywevS0d3tKJyjdbQZZqZVaiuAjiU/view'
+    # into 'https://studio.code.org/projects/flappy'
```

Comment:
I think we'd want this to be `https://studio.code.org/projects/flappy/view`, right? To distinguish it from the `/edit` path, and others we may have.

Reply:
done. See updated regex